### PR TITLE
add SkeletonPage component

### PR DESCRIPTION
**Pull request by chienit3bk**

### CODE REVIEW

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
+<template lang="pug">
+div(
+  :class="className",
+  role="status",
+  aria-label="{{lang['Polaris.SkeletonPage.loadingLabel']}}",
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:5]: Why do you need `{{`??

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+)
+  div(
+    :class="styles.Header",
+  )
+    div(
+      v-if="props.breadcrumbs",
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:11]: no need `props`

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    )
+      SkeletonBodyText(:lines="1")
+    div(
+      :class="styles.TitleAndPrimaryAction",
+    )
+      div(
+        :class="styles.TitleWrapper",
+      )
+        h1(
+          v-if="props.title",
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:23]: no need `props`

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          :class="styles.Title",
+        ) {{ props.title }}
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:25]: `props`

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        div(
+          v-else,
+          :class="styles.SkeletonTitle",
+        )
+      div(
+        v-if="props.primaryAction",
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:31]: `props`

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        :class="styles.PrimaryAction",
+      )
+        SkeletonDisplayText(:size="`large`")
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:34]: try `size="large"`

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  div(
+    :class="styles.Content",
+  )
+    slot
+</template>
+<script setup lang="ts">
+import styles from '@/classes/SkeletonPage.json';
+import { classNames } from 'polaris-react/src/utilities/css';
+import lang from 'polaris-react/locales/en.json';
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:43]: try using `inject` ( fetch `dev` to see reference on `Tooltip` component )

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+import { SkeletonBodyText, SkeletonDisplayText} from "@/components";
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:44]: lint

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  /** Remove the normal max-width on the page */
+  fullWidth?: boolean;
+  /** Decreases the maximum layout width. Intended for single-column layouts */
+  narrowWidth?: boolean;
+  /** Shows a skeleton over the primary action */
+  primaryAction?: boolean;
+  /** Shows a skeleton over the breadcrumb */
+  breadcrumbs?: boolean;
+}
+const props = withDefaults(defineProps<Props>(), {
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:58]: just `defineProps` is enough no need `default`

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        :class="styles.PrimaryAction",
+      )
+        SkeletonDisplayText(size="large")
+  div(
+    :class="styles.Content",
+  )
+    slot
+</template>
+<script setup lang="ts">
+import styles from '@/classes/SkeletonPage.json';
```
Comment (HQCuong) [inline src/components/SkeletonPage/SkeletonPage.vue:41]: reorder import codes

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  primaryAction?: boolean;
+  /** Shows a skeleton over the breadcrumb */
+  breadcrumbs?: boolean;
+}
+
+const props = defineProps<Props>();
+
+const lang = inject('lang') as Record<string, any>;
+
+const className = classNames(
```
Comment (HQCuong) [inline src/components/SkeletonPage/SkeletonPage.vue:63]: Should be computed

Comment (chienit3bk) [reply src/components/SkeletonPage/SkeletonPage.vue:63]: but it is static text

Comment (blacklen) [reply src/components/SkeletonPage/SkeletonPage.vue:63]: It's using `props` to render value. So it needs to be `reactive`

File: src/components/SkeletonPage/SkeletonPage.vue [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+  primaryAction?: boolean;
+  /** Shows a skeleton over the breadcrumb */
+  breadcrumbs?: boolean;
+}
+
+const props = defineProps<Props>();
+
+const lang = inject('lang') as Record<string, any>;
+
+const className = computed(()=>classNames(
```
Comment (blacklen) [inline src/components/SkeletonPage/SkeletonPage.vue:63]: lint

**Review by blacklen — APPROVED** [review]

**Pull request merged**